codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -352,13 +376,9 @@ private static void validateChar(char character) {
 
         @Override
         public CharSequence convertObject(Object value) {
-            checkNotNull(value, "value");
             if (value instanceof CharSequence) {
                 return (CharSequence) value;
             }
-            if (value instanceof Number) {
```

Comment:
Just pruning code a bit. `toString()` is already done at the end of the block.  I can revert if you like (if you think the slightly faster path for `Number` at the expense of other types is justified).